src/accelerate/commands/estimate.py

```diff
@@ -0,0 +1,270 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+
+from accelerate import init_empty_weights
+from accelerate.utils import (
+    calculate_maximum_sizes,
+    convert_bytes,
+    is_huggingface_hub_available,
+    is_timm_available,
+    is_transformers_available,
+)
+
+
+if is_huggingface_hub_available():
+    from huggingface_hub import model_info
+    from huggingface_hub.utils import GatedRepoError, RepositoryNotFoundError
+
+if is_transformers_available():
+    import transformers
+    from transformers import AutoConfig, AutoModel
+
+if is_timm_available():
+    import timm
+
+
+def verify_on_hub(repo: str):
+    "Verifies that the model is on the hub and returns the model info."
+    if not is_huggingface_hub_available():
+        raise ImportError(
+            "To use `accelerate estimate-memory`, the `huggingface_hub` library must be installed. Please run `pip install huggingface_hub` and try again."
+        )
+    try:
+        return model_info(repo)
+    except GatedRepoError:
+        return "gated"
+    except RepositoryNotFoundError:
+        return "repo"
+
+
+def check_has_model(error):
+    """
+    Checks what library spawned `error` when a model is not found
+    """
+    if is_timm_available() and isinstance(error, RuntimeError) and "Unknown model" in error.args[0]:
+        return "timm"
+    elif (
+        is_transformers_available()
+        and isinstance(error, OSError)
+        and "does not appear to have a file named" in error.args[0]
+    ):
+        return "transformers"
+    else:
+        return "unknown"
+
+
+def create_empty_model(model_name, library_name: str, trust_remote_code: bool = False):
+    """
+    Creates an empty model from its parent library on the `Hub` to calculate the overall memory consumption.
+
+    Args:
+        model_name (`str`):
+            The model name on the Hub
+        library_name (`str`):
+            The library the model has an integration with, such as `transformers`. Will be used if `model_name` has no
+            metadata on the Hub to determine the library.
+        trust_remote_code (`bool`, `optional`, defaults to `False`):
+            Whether or not to allow for custom models defined on the Hub in their own modeling files. This option
+            should only be set to `True` for repositories you trust and in which you have read the code, as it will
+            execute code present on the Hub on your local machine.
+
+    """
+    model_info = verify_on_hub(model_name)
+    # Simplified errors
+    if model_info == "gated":
+        raise GatedRepoError(f"Repo for model `{model_name}` is gated. You must be authenticated to access it.")
```

Comment:
Tell the user how to be authenticated (probably running `huggingface-cli login` or something like that)